src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -128,13 +128,26 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddSingleton<IFunctionMetadataManager, FunctionMetadataManager>();
             services.AddSingleton<IFunctionMetadataProvider, HostFunctionMetadataProvider>();
             services.AddSingleton<IWebFunctionsManager, WebFunctionsManager>();
-            services.AddSingleton<IInstanceManager, InstanceManager>();
             services.AddHttpClient();
             services.AddSingleton<StartupContextProvider>();
             services.AddSingleton<IFileSystem>(_ => FileUtility.Instance);
             services.AddTransient<VirtualFileSystem>();
             services.AddTransient<VirtualFileSystemMiddleware>();
 
+            if (SystemEnvironment.Instance.IsLinuxConsumptionOnAtlas())
```

Comment:
This statement is not needed if the `else` statement will perform the same logic. We can boil it down to:

``` CSharp
if (SystemEnvironment.Instance.IsLinuxConsumptionOnLegion())
{
    services.AddSingleton<IInstanceManager, LegionInstanceManager>();
 }
else
{
    // Default IInstanceManager
    services.AddSingleton<IInstanceManager, AtlasInstanceManager>();
}
```